pdr_backend/sim/sim_engine.py

```diff
@@ -261,3 +291,113 @@ def save_state(self, i: int, N: int):
             return False, False
 
         return True, False
+
+    @enforce_types
+    def _get_predictions_signals_data(
+        self, start_slot: int, end_slot: int
+    ) -> Dict[int, Optional[float]]:
+        result_dict: Dict[int, Optional[float]] = {}
+        db = DuckDBDataStore(self.ppss.lake_ss.lake_dir)
+        query_cont = f"""
+            SELECT
+                contract
+            FROM
+                pdr_predictions
+            WHERE
+                timeframe = '{self.predict_feed.timeframe}'
+                AND pair = '{self.predict_feed.pair.pair_str}'
+            LIMIT 1
+        """
+        feed_contract_addr = db.query_data(query_cont)
+        if len(feed_contract_addr) == 0:
```

Comment:
if the result is a list, you can use `if not feed_contract_addr`

Reply:
also, if it is a list, I suggest `feed_contract_addrs` (plural)

Reply:
it's just 1 contract, but the result is a DataFrame that's why it is checked with len()